src/CowEvcClosePositionWrapper.sol

```diff
@@ -0,0 +1,277 @@
+    function encodePermitData(ClosePositionParams memory params) external view returns (bytes memory) {
+        (IEVC.BatchItem[] memory items,) = _encodeBatchItemsAfter(memoryLocation(params));
+        return _encodePermitData(items, memoryLocation(params));
+    }
+
+    function _encodeBatchItemsAfter(ParamsLocation paramsLocation)
+        internal
+        view
+        override
+        returns (IEVC.BatchItem[] memory items, bool needsPermission)
+    {
+        ClosePositionParams memory params = paramsFromMemory(paramsLocation);
+        items = new IEVC.BatchItem[](1);
+
+        // 1. Repay debt and return remaining assets
+        items[0] = IEVC.BatchItem({
+            onBehalfOfAccount: params.account,
+            targetContract: address(this),
+            value: 0,
+            data: abi.encodeCall(this.helperRepay, (params.borrowVault, params.owner, params.account))
+        });
+
+        needsPermission = true;
+    }
+
+    /// @notice Called by the EVC after a CoW swap is completed to repay the user's debt. Will use all available collateral in the user's account to do so.
+    /// @param vault The Euler vault in which the repayment should be made
+    /// @param owner The owner associated with the given account. This the owner is used to provide the funds for the repayment.
+    /// @param account The subaccount that should be receiving the repayment of debt
+    function helperRepay(address vault, address owner, address account) external {
+        require(msg.sender == address(EVC), Unauthorized(msg.sender));
+        (address onBehalfOfAccount,) = EVC.getCurrentOnBehalfOfAccount(address(0));
+        require(onBehalfOfAccount == account, Unauthorized(onBehalfOfAccount));
+
+        // Subaccounts in the EVC can be any account that shares the highest 19 bits as the owner.
+        // Here we verify that the subaccount address has been specified as expected.
+        // This is a really important security check because without verifying the subaccount,
+        // any user could potentially repay
+        require(bytes19(bytes20(owner)) == bytes19(bytes20(account)), SubaccountMustBeControlledByOwner(account, owner));
+
+        IERC20 asset = IERC20(IERC4626(vault).asset());
+
+        uint256 debtAmount = IBorrowing(vault).debtOf(account);
+
+        // repay as much debt as we can
+        uint256 repayAmount = asset.balanceOf(owner);
+        if (repayAmount > debtAmount) {
+            // the user intends to repay all their debt. we will revert if their balance is not sufficient.
+            repayAmount = debtAmount;
+        }
+
+        // pull funds from the user (they should have approved spending by this contract)
+        SafeERC20Lib.safeTransferFrom(asset, owner, address(this), repayAmount, address(0));
+
+        // repay what was requested on the vault
+        asset.approve(vault, repayAmount);
+        IBorrowing(vault).repay(repayAmount, account);
```

Comment:
`repay` returns a uint256, can it be that the debt is capped and the function doesn't take the full `repayAmount`, leaving funds in the contract (and an open allowance)?
In case, we should send back the difference and zero the allowance.

I see though that this is taken into account here: you cap using ` IBorrowing(vault).debtOf(account);`, while `repay` in the EVC caps with ~`getCurrentOwed`. What's the difference?
In general, all these considerations should be added as a code comment since they seem very relevant to the logic.